pages/api/handoff.page.ts

```diff
@@ -36,13 +37,17 @@ export const returnRedirectUrl = async (
     >({
       query: GetDefaultAccountDocument,
     });
-    let defaultAccountID = req.query?.accountListId;
-    if (!defaultAccountID) {
-      const {
-        data: { user, accountLists },
-      } = response;
-      defaultAccountID = user?.defaultAccountList || accountLists?.nodes[0]?.id;
-    }
+    const {
+      data: { user, accountLists },
+    } = response;
+
+    const defaultAccountID =
+      isImpersonating && user.defaultAccountList
```

Comment:
Do we want to fall back to `accountLists?.nodes[0]?.id` too even when impersonating?

Reply:
That's a great point